feat: flutter alpha preparation (#1684)
ZupIT/beagle: flutter/readme.md

## Versioning
Every alpha and beta version of Beagle Flutter will follow the pattern `0.x.y`, where `x` is the version of Beagle Web
it's based on and `y` is every subsequent version where `x` would be the same.

In the version number, `x` refers to a version of the Beagle Web, because Beagle Flutter uses this lib under the hood.

## Null safety compatibility
For now, we'll not support null safety for the alpha versions, however this is a high priority implementation and 
should be available in a beta version of Beagle.

## Installation
1. Open the file `pubspec.yaml` in the root of your project.
2. Under `dependencies`, add `beagle: ^0.0.9`, or whatever the most recent version is.
3. Also add `beagle_components: ^0.0.9`. You can omit this dependency if you're familiar with Beagle and won't use any
of the default components.
4. From your IDE, click `pub get`. Or, from the terminal, type `flutter pub get`.

## Usage
We currently don't have a script to automatically setup your environment, but don't be discouraged, it's quite simple.

### 1. The configuration
All the configuration necessary for Beagle to work is centered on the parameters of the `BeagleSdk.init` startup 

> method. This params tells everything Beagle needs to know to render your widgets. Here we show only the basic options 

Author: In case I'm referring to the method's parameters, then it's in the plural.

implementations within the library. This is probably not going to be the case in the future, since it may represent
potential security issues. The Beagle services are:

- [BeagleLogger](todo): provide a logger for Beagle to use.
- [HttpClient](todo): tell exactly how the network requests should be made, add your own headers and much more.
- [BeagleImageDownloader](todo): use your own logic for downloading network images.
- [BeagleStorage](todo): persist data across multiple executions by stating exactly how Beagle should store data.

You can set your own implementation of each of these services via `BeagleSdk.init`, in your project.

## Customization
Beagle is highly customizable, you can create your own components, actions and even operations. All of these must be
provided when calling `BeagleSdk.init`.

- [Custom components](todo): create your own components.
- [Custom actions](todo): make the events in your components do exactly what you need.
- [Custom operations](todo): if the operations shipped with Beagle are not enough for your expressions, create your own.
- [Design System](todo): here you define all styles and local images that can be used by Beagle.

> 

Reviewer: I think the description above, of Custom components, actions and operations are redundant. It could be omitted

Author: I think it's positive to let explicit what the user can customize.

## Other APIs

> - [Global context](todo): learn how to manipulate the global context in Beagle Flutter.

Reviewer: ```suggestion
- [Global context](todo): allows manipulation of the global context in Beagle Flutter.
```

> - [Analytics](todo): observe every action executed and navigation performed.

Reviewer: ```suggestion
- [Analytics](todo): gives information of every action executed, such as navigation data.
```

- [Renderer](todo): sometimes it might be necessary to interact with Beagle while rendering a component. This article
  shows how to use the Renderer API to achieve complex behaviors.
  
## Current state of development

> We're currently in alpha. We'll move to beta as soon as we have every layout tool (Yoga) working as expected. Here's

Reviewer: ```suggestion
Currently in alpha. It'll be moved to beta as soon as we have every layout tool (Yoga) working as expected. Here's
```
